File: scripts/verify_log_view.py
"""启本地服务 → playwright 截图 LOG 视图 → 退出。"""
import os, sys, time, subprocess, threading, http.server, socketserver, functools

ROOT = os.path.dirname(os.path.abspath(__file__))
DASH = os.path.join(ROOT, "..", "dashboard")
PORT = 8788  # 避开可能占用
OUT_DIR = os.path.join(ROOT, "..", "artifacts")
os.makedirs(OUT_DIR, exist_ok=True)

# 启 server
os.chdir(DASH)
handler = functools.partial(http.server.SimpleHTTPRequestHandler, directory=DASH)
httpd = socketserver.TCPServer(("127.0.0.1", PORT), handler)
threading.Thread(target=httpd.serve_forever, daemon=True).start()
print(f"server up: http://127.0.0.1:{PORT}/")

from playwright.sync_api import sync_playwright

with sync_playwright() as pw:
    browser = pw.chromium.launch(headless=True, args=["--no-sandbox"])
    ctx = browser.new_context(viewport={"width": 1440, "height": 2400}, device_scale_factor=1.5)
    page = ctx.new_page()
    page.goto(f"http://127.0.0.1:{PORT}/index.html", wait_until="networkidle", timeout=30000)
    page.wait_for_timeout(1500)  # 开机序列
    # 切到 LOG 视图
    page.evaluate("show('LOG')")
    page.wait_for_timeout(800)
    # 1) 完整 LOG 区块全屏
    page.screenshot(path=os.path.join(OUT_DIR, "dash_LOG_full.png"), full_page=True)
    # 2) 仅 LOG 视图
    el = page.locator("#view-LOG")
    el.screenshot(path=os.path.join(OUT_DIR, "dash_LOG_view.png"))
    # 3) 排名 + 协议 + 结论区
    page.locator("#view-LOG .panel").screenshot(path=os.path.join(OUT_DIR, "dash_LOG_panel.png"))
    # 4) RUN HISTORY 表格应用一次筛选：按"blind-holdout"
    page.select_option("#fProto", "blind-holdout")
    page.wait_for_timeout(300)
    el.screenshot(path=os.path.join(OUT_DIR, "dash_LOG_filter_holdout.png"))
    # 5) 排序演示：点收益降序
    page.select_option("#fProto", "")
    page.wait_for_timeout(200)
    page.click('th[data-sort="total_return_pct"]')
    page.click('th[data-sort="total_return_pct"]')  # 第二次降序
    page.wait_for_timeout(200)
    page.locator("#view-LOG").screenshot(path=os.path.join(OUT_DIR, "dash_LOG_sorted.png"))
    print("screenshots saved to", OUT_DIR)
    browser.close()

httpd.shutdown()

File: scripts/verify_log_view.js
// 启本地服务 + puppeteer-core 截图 LOG 视图
const { spawn } = require('child_process');
const path = require('path');
const http = require('http');
const fs = require('fs');

const ROOT = path.resolve(__dirname, '..', 'dashboard');
const OUT = path.resolve(__dirname, '..', 'artifacts');
fs.mkdirSync(OUT, { recursive: true });

const PORT = 8788;
const server = spawn('python3', ['-m', 'http.server', String(PORT), '--directory', ROOT, '--bind', '127.0.0.1'], {
  stdio: 'ignore'
});

function waitReady(cb) {
  const tick = () => http.get({ host: '127.0.0.1', port: PORT, path: '/' }, res => {
    res.resume(); cb();
  }).on('error', () => setTimeout(tick, 200));
  setTimeout(tick, 200);
}

async function shotEl(page, sel, outName) {
  const box = await page.evaluate((s) => {
    const el = document.querySelector(s);
    if (!el) return null;
    const r = el.getBoundingClientRect();
    return { x: r.x, y: r.y, width: r.width, height: r.height };
  }, sel);
  if (!box) { console.log('miss', sel); return; }
  await page.screenshot({
    path: path.join(OUT, outName),
    clip: { x: box.x, y: box.y, width: box.width, height: Math.min(box.height, 4000) }
  });
  console.log('shot', outName, box.width.toFixed(0), 'x', Math.min(box.height, 4000).toFixed(0));
}

(async () => {
  await new Promise(r => waitReady(r));
  console.log('server up');
  const puppeteer = require('/home/gem/.npm-global/lib/node_modules/puppeteer-core');
  const browser = await puppeteer.launch({
    executablePath: '/opt/chromium.org/chromium/chromium-browser',
    headless: 'new',
    args: ['--no-sandbox', '--disable-dev-shm-usage']
  });
  const page = await browser.newPage();
  await page.setViewport({ width: 1440, height: 900, deviceScaleFactor: 1.5 });
  await page.goto(`http://127.0.0.1:${PORT}/index.html`, { waitUntil: 'networkidle0', timeout: 30000 });
  await new Promise(r => setTimeout(r, 1800));
  await page.evaluate(() => { show('LOG'); });
  await new Promise(r => setTimeout(r, 700));

  await page.screenshot({ path: path.join(OUT, 'dash_LOG_full.png'), fullPage: true });
  await shotEl(page, '#view-LOG .panel', 'dash_LOG_panel.png');

  await page.select('#fProto', 'blind-holdout');
  await new Promise(r => setTimeout(r, 300));
  await shotEl(page, '#view-LOG', 'dash_LOG_filter_holdout.png');

  await page.click('#fReset');
  await new Promise(r => setTimeout(r, 200));
  await page.click('th[data-sort="total_return_pct"]');
  await page.click('th[data-sort="total_return_pct"]');
  await new Promise(r => setTimeout(r, 200));
  await shotEl(page, '#view-LOG', 'dash_LOG_sorted.png');

  await page.evaluate(() => { show('BTCUSDT'); });
  await new Promise(r => setTimeout(r, 1200));
  await page.screenshot({ path: path.join(OUT, 'dash_BTC_home.png'), fullPage: false });

  await page.evaluate(() => { show('SYSTEM'); });
  await new Promise(r => setTimeout(r, 500));
  await shotEl(page, '#view-SYSTEM', 'dash_SYSTEM.png');

  console.log('OK');
  await browser.close();
  server.kill();
  process.exit(0);
})().catch(e => { console.error(e); server.kill(); process.exit(1); });
